lib/data_hygiene/policy_publication_history_writer.rb

```diff
@@ -0,0 +1,102 @@
+module DataHygiene
+  class PolicyPublicationHistoryWriter
+    def initialize(publication, policy, logger = NullLogger.instance)
+      @publication = publication
+      @policy = policy
+      @logger = logger
+    end
+
+    def rewrite_history!
+      Edition::AuditTrail.acting_as(gds_user) do
```

Comment:
There's a transaction wrapping the entire thing in the script where this is ultimately used: https://github.com/alphagov/whitehall/pull/2195/files#diff-592485b47290fc266306ad284ceeededR23